### PR TITLE
Member Avatar & Status [MAS] - Dev-3.0.2

### PR DESCRIPTION
**Pull request by dark-1**
Waiting for : 
- [x] https://github.com/phpbb/phpbb/pull/5572
- [x] https://github.com/phpbb/phpbb/pull/5573
- [x] https://github.com/phpbb/phpbb/pull/5574
- [x] https://github.com/phpbb/phpbb/pull/5634
- [x] https://github.com/phpbb/phpbb/pull/5642
- [x] https://github.com/phpbb/phpbb/pull/5643
- [x] Fix what **3D-I** pointed out.
- [x] Fix whole Repository.
- [x] Wait for phpBB v3.2.8 release.

Will Merge when All of the above are Done.

### CODE REVIEW

File: dark1/memberavatarstatus/adm/style/acp_mas_main.html [hunk 1]
```diff
@@ -99,7 +99,112 @@ <h2>{{ lang('ACP_MAS_MODE_MAIN') }}</h2>
...
+								<div id="dark1_mas_sh_fp_av" class="toggleWrapper {% if MAS_SH_FP_AVATAR %}toggleChecked{% endif %}" >
+									<span for="dark1_mas_sh_fp_av" class="toggle">
+										<span class="toggle_handle"></span>
+									</span>
+								</div>
+							</dd>
+						</dl>
+						<dl>
+							<dt>
+								<label>{{ lang('ACP_MAS_SH_FP_AV_SZ_LB') }}{{ lang('COLON') }}</label>
```
Comment (3D-I) [inline dark1/memberavatarstatus/adm/style/acp_mas_main.html]: `{{ lang('ACP_MAS_SH_FP_AV_SZ_LB') ~ lang('COLON') }}` is better concatenated.
Same applies for all of the occurencies. from now on.

File: dark1/memberavatarstatus/adm/style/acp_mas_main.html [hunk 2]
```diff
@@ -99,7 +99,112 @@ <h2>{{ lang('ACP_MAS_MODE_MAIN') }}</h2>
...
+							</dt>
+							<dd>
+								<span class="numSize"><span class="numVal">{{ MAS_SH_FP_AV_SIZE }}</span>&nbsp;{{ lang('ACP_MAS_AV_SIZE_PX') }}</span>
```
Comment (3D-I) [inline dark1/memberavatarstatus/adm/style/acp_mas_main.html]: classes should be lowercased.

File: dark1/memberavatarstatus/adm/style/acp_mas_review.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
+{% INCLUDE 'overall_header.html' %}
+{% INCLUDECSS '@dark1_memberavatarstatus/mas_acp_style.css' %}
+{% INCLUDEJS '@dark1_memberavatarstatus/mas_acp_script.js' %}
+
+<h1>{{ MAS_EXT_NAME }}</h1>
+<div>{{ lang('ACP_MAS_BY') }}&nbsp;{{ MAS_EXT_DEV }}</div>
+<hr style="margin: 5px 0px 0px 0px; padding: 5px 0px 0px 0px; border-top: 2px solid #888;" />
```
Comment (3D-I) [inline dark1/memberavatarstatus/adm/style/acp_mas_review.html]: avoid inline style if possible

File: dark1/memberavatarstatus/adm/style/acp_mas_review.html [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+			</dd>
+		</dl>
+		<dl>
+			<dt>
+				<label for="dark1_mas_rv_av_sz">{{ lang('ACP_MAS_RV_AV_SZ_LB') }}{{ lang('COLON') }}</label>
+				<br />
+				<div>{{ lang('ACP_MAS_AV_SZ_SML_EXPLAIN') }}</div>
+			</dt>
+			<dd>
+				<input type="number" min="9" max="99" size="3" maxlength="3" step="1" name="dark1_mas_rv_av_sz" value="{{ MAS_RV_AV_SIZE }}" oninput="sync_avatar_size('dark1_mas_rv_av_sz','div_dark1_mas_rv_av_sz')" id="dark1_mas_rv_av_sz" />&nbsp;{{ lang('ACP_MAS_AV_SIZE_PX') }}
```
Comment (3D-I) [inline dark1/memberavatarstatus/adm/style/acp_mas_review.html]: Avoid `&nbsp;` inline but use a normal space.

File: dark1/memberavatarstatus/adm/style/acp_mas_review.html [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+				<br />
+				<br />
+				<div style="overflow: auto; max-height: 300px; max-width: 300px;">
```
Comment (3D-I) [inline dark1/memberavatarstatus/adm/style/acp_mas_review.html]: no inline style but CSS

File: dark1/memberavatarstatus/core/memberavatarstatus.php [hunk 6]
```diff
@@ -30,7 +30,7 @@ class memberavatarstatus
 	const AV_MIN_SZ = 9;
```
Comment (3D-I) [inline dark1/memberavatarstatus/core/memberavatarstatus.php]: constants should go preferably into the ext.php file.

Then `use vendor/name/ext;`

File: dark1/memberavatarstatus/event/listener.php [hunk 7]
```diff
@@ -77,26 +77,31 @@ static public function getSubscribedEvents()
 			// Main Setup
```
Comment (3D-I) [inline dark1/memberavatarstatus/event/listener.php]: Too much of events in only one listener.
All this should be divided into localized listeners by purpose.

File: dark1/memberavatarstatus/event/listener.php [hunk 8]
```diff
@@ -169,6 +174,10 @@ public function mas_header($event)
 			'MAS_SH_UP_AVATAR'	=> $this->config['dark1_mas_sh_up_av'],
 			'MAS_SH_UP_AV_SIZE'	=> $this->mas_func->mas_get_config_avatar_size('dark1_mas_sh_up_av_sz', memberavatarstatus::AV_DEF_SZ_SML, memberavatarstatus::AV_MAX_SZ_SML),
 			'MAS_SH_UP_ONLINE'	=> $this->config['dark1_mas_sh_up_ol'],
```
Comment (3D-I) [inline dark1/memberavatarstatus/event/listener.php]: cast to (bool) each configuration that has a yes or no as a question.

File: dark1/memberavatarstatus/language/en/lang_acp_mas.php [hunk 9]
```diff
@@ -73,43 +73,51 @@
...
+	'ACP_MAS_SH_UP_OL_LB'		=> 'Online Status Enable',
 	'ACP_MAS_SH_UP_OL_EXPLAIN'	=> 'Enables the Post User Online Status in Search,<br />That is in All Post Search Pages.<br />Default : No',
+	// Review
+	'ACP_MAS_RV_AV_SET'			=> 'Avatar Setting',
+	'ACP_MAS_RV_AV_LB'			=> 'Avatar Enable',
+	'ACP_MAS_RV_AV_EXPLAIN'		=> 'Enables the Avatar in Review Block,<br />That is in “Posting Topic Review” , “MCP Topic Review” & “PM History Review” Page.<br />Default : No',
+	'ACP_MAS_RV_AV_SZ_LB'		=> 'Avatar Size',
+	'ACP_MAS_RV_OL_SET'			=> 'Online Status Setting',
+	'ACP_MAS_RV_OL_LB'			=> 'Online Status Enable',
+	'ACP_MAS_RV_OL_EXPLAIN'		=> 'Enables the Online Status in Review Block,<br />That is in “Posting Topic Review” , “MCP Topic Review” & “PM History Review” Page.<br />Default : No',
```
Comment (3D-I) [inline dark1/memberavatarstatus/language/en/lang_acp_mas.php]: `<br>` instead of `<br />`

**Comment by dark-1** [comment]
Thanks @3D-I , I'll do the needful.

**Comment by dark-1** [comment]
@3D-I ,

can you scrutinize this whole branch : https://github.com/dark-1/memberavatarstatus/tree/dev-3.x.y
If you have time can you help me pointing out my mistakes, since many of your ext's are accepted...
coz approval of this ext have failed plenty of time , so at least while submitting this time would be accepted , 

Thanking you , best regards 👍

File: dark1/memberavatarstatus/adm/style/acp_mas_main.html (hunk 1 above)
Comment (dark-1) [reply dark1/memberavatarstatus/adm/style/acp_mas_main.html]: Done 👍

File: dark1/memberavatarstatus/adm/style/acp_mas_main.html (hunk 2 above)
Comment (dark-1) [reply dark1/memberavatarstatus/adm/style/acp_mas_main.html]: Done 👍

File: dark1/memberavatarstatus/adm/style/acp_mas_review.html (hunk 3 above)
Comment (dark-1) [reply dark1/memberavatarstatus/adm/style/acp_mas_review.html]: Done 👍

File: dark1/memberavatarstatus/adm/style/acp_mas_review.html (hunk 4 above)
Comment (dark-1) [reply dark1/memberavatarstatus/adm/style/acp_mas_review.html]: But , I prefer `&nbsp;` over ` ` , coz it's better than normal space.
Also , no one in extension validation team pointed that out...

File: dark1/memberavatarstatus/adm/style/acp_mas_review.html (hunk 5 above)
Comment (dark-1) [reply dark1/memberavatarstatus/adm/style/acp_mas_review.html]: Done 👍

File: dark1/memberavatarstatus/core/memberavatarstatus.php (hunk 6 above)
Comment (dark-1) [reply dark1/memberavatarstatus/core/memberavatarstatus.php]: I was told by extension validation team to put it in core file ie `./core/memberavatarstatus.php` .

File: dark1/memberavatarstatus/language/en/lang_acp_mas.php (hunk 9 above)
Comment (dark-1) [reply dark1/memberavatarstatus/language/en/lang_acp_mas.php]: XHTML compliant

File: dark1/memberavatarstatus/event/listener.php (hunk 7 above)
Comment (dark-1) [reply dark1/memberavatarstatus/event/listener.php]: Done 👍

**Pull request merged**